### PR TITLE
chore: Integrate solhint with a few rules to enforce contract convention

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #

# Integrates solhint

https://github.com/ubiquity/ubiquity-dollar/assets/41552663/3836a9c5-088a-4882-8244-79e3537ee247

### CODE REVIEW

**Comment by molecula451** [comment]
thoughts on having .solhint as one of the chosen linters for solidity? @rndquu @pavlovcik @gitcoindev

**Comment by molecula451** [comment]
i think if we agree we can have the package already included, and selectively build these rules

**Comment by 0x4007** [comment]
> thoughts on having .solhint as one of the chosen linters for solidity? @rndquu @pavlovcik @gitcoindev 

I did limited research on the matter but 1. I recognize the package name as being pretty popular and 2. If it works then my vote is let's use it

---

Usually I review GitHub notifications from my phone. For some reason on both my browser and the GitHub app, videos don't load so I don't have all the context.

**Comment by molecula451** [comment]
> > thoughts on having .solhint as one of the chosen linters for solidity? @rndquu @pavlovcik @gitcoindev
> 
> I did limited research on the matter but 1. I recognize the package name as being pretty popular and 2. If it works then my vote is let's use it
> 
> Usually I review GitHub notifications from my phone. For some reason on both my browser and the GitHub app, videos don't load so I don't have all the context.

ahhh.. the well known video Github app issue, out of context: (i wonder why they don't fix it, it still happens, lol)

dropboxed to you @pavlovcik : https://www.dropbox.com/scl/fi/qcmrhzg04zrqvdw97syni/solhint.mp4?rlkey=cvli23m2am7x0ekxidf6um0qk&dl=0

**Comment by 0x4007** [comment]
> > > thoughts on having .solhint as one of the chosen linters for solidity? @rndquu @pavlovcik @gitcoindev
> > 
> > 
> > I did limited research on the matter but 1. I recognize the package name as being pretty popular and 2. If it works then my vote is let's use it
> > Usually I review GitHub notifications from my phone. For some reason on both my browser and the GitHub app, videos don't load so I don't have all the context.
> 
> ahhh.. the well known video Github app issue, out of context: (i wonder why they don't fix it, it still happens, lol)
> 
> dropboxed to you @pavlovcik : https://www.dropbox.com/scl/fi/qcmrhzg04zrqvdw97syni/solhint.mp4?rlkey=cvli23m2am7x0ekxidf6um0qk&dl=0

GitHub iOS ironically is not open sourced. I tried researching and implementing a fix already 👀

File: packages/contracts/package.json [hunk 1]
```diff
@@ -49,7 +49,8 @@
     "dollar",
     "decentralization",
     "token",
-    "algorithmic"
+    "algorithmic",
+    "solidity"
```
Comment (0x4007) [inline packages/contracts/package.json]: Solidity is capitalized, perhaps it is proper to remove this?

Comment (molecula451) [reply packages/contracts/package.json]: yeah will remove this, is not part of the integration anyway

Comment (gitcoindev) [reply packages/contracts/package.json]: I also think this can be removed

**Comment by rndquu** [comment]
I'm ok to use solhint. It's [security rules](https://github.com/protofire/solhint/blob/develop/docs/rules.md#security-rules) are pretty outdated (slither works much better) but [style guide](https://github.com/protofire/solhint/blob/develop/docs/rules.md#style-guide-rules) and [best practices](https://github.com/protofire/solhint/blob/develop/docs/rules.md#best-practise-rules) rules could be useful.

**Review by gitcoindev — CHANGES_REQUESTED** [review]
It looks good from the integration side. I would add to `README.md` a short information on enabled solhint in https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md#committing-codesending-prs section .

**Comment by molecula451** [comment]
im mergin but not closing the issue #849 since this is only the ingration of the module, so the issue serves to craft further rules that'll we be using

**Comment by rndquu** [comment]
@molecula451 Could you add `.solhint.json` to exceptions [here](https://github.com/ubiquity/ubiquity-dollar/blob/f6bd4c961383a95ad5e91f44f280849763a8be84/.github/workflows/enforce-kebab-case.yml#L17) to make "Enforce kebab-case" workflow pass without errors?

**Comment by molecula451** [comment]
> @molecula451 Could you add `.solhint.json` to exceptions [here](https://github.com/ubiquity/ubiquity-dollar/blob/f6bd4c961383a95ad5e91f44f280849763a8be84/.github/workflows/enforce-kebab-case.yml#L17) to make "Enforce kebab-case" workflow pass without errors?

can you back me up on this one? if not i will PR in a bit

**Comment by gitcoindev** [comment]
> > @molecula451 Could you add `.solhint.json` to exceptions [here](https://github.com/ubiquity/ubiquity-dollar/blob/f6bd4c961383a95ad5e91f44f280849763a8be84/.github/workflows/enforce-kebab-case.yml#L17) to make "Enforce kebab-case" workflow pass without errors?
> 
> can you back me up on this one? if not i will PR in a bit

I opened https://github.com/ubiquity/ubiquity-dollar/pull/853

**Pull request merged**